rules_ruby_gapic/gapic-generator-ads/repositories.bzl

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Defines the function that will load the dependencies for gapic-generator-ads
+"""
+load("//rules_ruby_gapic/ruby:ruby_runtime.bzl", "ruby_runtime")
+load("//rules_ruby_gapic/ruby:bundle_install.bzl", "bundle_install")
+
+##
+# Creates a ruby runtime specific to gapic-generator-ads and 
+# a bundled installation for its packages
+#
+def gapic_generator_ads_repos():
+  # Create the ruby runtime
+  ruby_runtime(
```

Comment:
Runtime has to install a version of bundler specified in the lockfile. Ads and cloud do have differrent lockfiles and they can be bundled with different bundlers. Hence different runtimes.
Luckily with this change runtime is a much smaller commitment -- it does not install any gems except bundler after all.

Potential errors from shared runtime seem worse to me. 

At some point if we can add a git hook to keep the lockfiles in sync this can be changed altho I don't think that's a good idea either.